services/director-v2/src/simcore_service_director_v2/modules/dynamic_sidecar/volumes_resolver.py

```diff
@@ -82,30 +86,79 @@ def _volume_name(cls, path: Path) -> str:
         return f"{path}".replace(os.sep, "_")
 
     @classmethod
-    def source(cls, compose_namespace: str, path: Path) -> str:
-        return f"{compose_namespace}{cls._volume_name(path)}"
+    def source(cls, path: Path, node_uuid: NodeID, run_id: RunID) -> str:
+        """
+        Guarantees that the volume name is unique between runs while also
+        taking into consideration the limit for the volume name's length
+        (255 characters).
+
+        Example:
+        >>> source(
+            path="/this/is/a/data/folder",
+            node_uuid="17f5b46d-505c-489e-9818-f6d294c892d9",
+            run_id="a5d2a8bd-eaef-4d7b-8dd9-1ae4e408f3c3"
+        )
+        dyv_a5d2a8bd-eaef-4d7b-8dd9-1ae4e408f3c3_redlof_atad_a_si_siht_17f5b46d-505c-489e-9818-f6d294c892d9
```

Comment:
I think the example is not fitting the reality...
and what is: redlof_atad_a_si_siht?

Reply:
Fixed example and added a test for this